syntax = "proto3";

package protos.engines;

import "protos/engines/query_base.proto";
import "protos/literal.proto";
import "google/protobuf/wrappers.proto";
import "protos/base.proto";


message GlobalQueryOptions {
  repeated OpDescription op_descriptions = 1;
  repeated OpMapping op_mapping = 2;
  repeated LiteralTypeDescription literal_type_description = 3;
}

message ColumnOption {
  message IdOption {
    IdLiteral.Type type = 1;
    map<uint64, string> long_options = 2;
    map<string, string> string_options = 3;
  }

  string name = 1;
  string alias = 2;
  LiteralType type = 3;
  bool is_groupable = 5;
  IdOption id_option = 10;
}


message QueryOptions {
  repeated ColumnOption column_options = 1;
}

